### PR TITLE
Merge `Add Filter` Dialog changes

### PR DESCRIPTION
**Pull request by PaperFanz**
* fix padding around list widget
* swap list widget for tree widget
* fix segfault on adding empty widget

### CODE REVIEW

**Comment by PaperFanz** [comment]
Tagging on the features @EAkita requested in PR #9 for review

File: insitu/lib/insitu/filter.hpp [hunk 1]
```diff
@@ -88,19 +87,27 @@ public Q_SLOTS:
 
 };    // class FilterWatchdog
 
+typedef enum FilterProps
+{
+    setToImageSize = 1,
+    keepAspectRatio = 2,
+    lockFilterProperties = 4
```
Comment (AdamPettinger) [inline insitu/lib/insitu/filter.hpp:94]: Is there a reason for skipping?

File: insitu/lib/insitu/filter.hpp [hunk 2]
```diff
@@ -159,7 +159,7 @@ class Filter : public nodelet::Nodelet
         while (exitCond.wait_for(1ms) == std::future_status::timeout)
         {
             updateFilter(apply());
-            std::this_thread::sleep_for(10ms);
+            std::this_thread::sleep_for(15666us);
```
Comment (AdamPettinger) [inline insitu/lib/insitu/filter.hpp:162]: 15666?

File: insitu/src/filter_graphics_item.cpp [hunk 3]
```diff
@@ -7,18 +7,20 @@ namespace insitu
...
+    {
         setFlag(QGraphicsItem::ItemIsSelectable, true);
         setFlag(QGraphicsItem::ItemIsMovable, true);
         setFlag(QGraphicsItem::ItemSendsGeometryChanges, true);
     }
 
     img = QImage(1, 1, QImage::Format_RGBA8888);
-    
+
     // some jank shit, don't touch
```
Comment (AdamPettinger) [inline insitu/src/filter_graphics_item.cpp:24]: lol

File: insitu_plugins/src/Stats/Stats.cpp [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+{
+    settingsDialog = new StatsDialog(this);
+    setSize(QSize(200, 120));
+}
+
+void Stats::onDelete(void)
+{
+}
+
+void Stats::drawtorect(cv::Mat& mat, cv::Rect target, const std::string& str,
```
Comment (AdamPettinger) [inline insitu_plugins/src/Stats/Stats.cpp:33]: Let's remember to change this one to the shared library too ya?

File: insitu_plugins/src/Stats/Stats.cpp [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+            (int)((double)target.height * (scaley - scale) / scaley * 0.5);
+    cv::putText(
+        mat, str,
+        cv::Point(target.x + marginx, target.y + target.height - marginy), face,
+        scale, color, thickness, cv::LINE_AA, false);
+}
+
+const cv::Mat Stats::apply(void)
+{
+    std::this_thread::sleep_for(1s);
```
Comment (AdamPettinger) [inline insitu_plugins/src/Stats/Stats.cpp:56]: The `apply` function is called every image (30 hz) right? Are there ramifications for sleeping here and not returning quickly?

**Review by AdamPettinger — APPROVED** [review]
I gave this a pretty quick look-through... lots of files changed

Left a few questions otherwise looks good

File: insitu/lib/insitu/filter.hpp (hunk 1 above)
Comment (PaperFanz) [reply insitu/lib/insitu/filter.hpp:94]: This is for bit flags, 1 = 0b0001, 2 = 0b0010, 4 = 0b0100, etc

File: insitu/lib/insitu/filter.hpp (hunk 2 above)
Comment (PaperFanz) [reply insitu/lib/insitu/filter.hpp:162]: that combined with the 1ms wait above limits filter update rate to 60Hz

File: insitu_plugins/src/Stats/Stats.cpp (hunk 5 above)
Comment (PaperFanz) [reply insitu_plugins/src/Stats/Stats.cpp:56]: Apply runs asynchronously from the main thread so sleeping/waiting within the filter does not affect the main GUI, your specific filter just won't update as quickly.

**Pull request merged**